File: engine-monsters/ransor.py

# ransor.py

class Ransor:
    def __init__(self):
        self.name = "Ransor"
        self.origin = "Muspelheim"
        self.buffs = {
            "Slayer": {"Void Essence": 1},
            "Juggernaut": {"Stamina Regen": 0.15},
            "Bruiser": {"Berzerker Boost": 0.25}
        }
        self.duration = 90
        self.respawn_time = 180

    def on_death(self):
        return "AoE Fear + Silence (1s)"

File: engine-monsters/Ransor.java

// Ransor.java

public class Ransor {
    private final String name = "Ransor";
    private final String origin = "Muspelheim";
    private final int duration = 90;
    private final int respawnTime = 180;

    public String getBuff(String championClass) {
        switch (championClass) {
            case "Slayer":
                return "+1 Void Essence";
            case "Juggernaut":
                return "+15% Stamina Regen";
            case "Bruiser":
                return "+25% Berzerker Boost";
            default:
                return "Partial Buff (50%)";
        }
    }

    public String onDeath() {
        return "AoE Fear + Silence (1s)";
    }
}
